crates/core/tedge_mapper/src/sm_c8y_mapper/topic.rs

```diff
@@ -1,83 +1,99 @@
-            Self::SoftwareUpdateResponse => r#"tedge/commands/res/software/update"#,
             Self::SmartRestRequest => r#"c8y/s/ds"#,
-            Self::RestartResponse => r#"tedge/commands/res/control/restart"#,
+            Self::SmartRestResponse => r#"c8y/s/us"#,
+        }
+    }
+
+    pub fn to_topic(&self) -> Result<Topic, MqttClientError> {
+        match self {
+            Self::SmartRestRequest => Topic::new(Self::SmartRestRequest.as_str()),
+            Self::SmartRestResponse => Topic::new(Self::SmartRestResponse.as_str()),
         }
     }
 }
 
-impl TryFrom<String> for IncomingTopic {
-    type Error = MapperTopicError;
+impl TryFrom<String> for C8yTopic {
+    type Error = TopicError;
 
     fn try_from(value: String) -> Result<Self, Self::Error> {
         match value.as_str() {
-            r#"tedge/commands/res/software/list"# => Ok(IncomingTopic::SoftwareListResponse),
-            r#"tedge/commands/res/software/update"# => Ok(IncomingTopic::SoftwareUpdateResponse),
-            r#"c8y/s/ds"# => Ok(IncomingTopic::SmartRestRequest),
-            r#"tedge/commands/res/control/restart"# => Ok(IncomingTopic::RestartResponse),
-            err => Err(MapperTopicError::UnknownTopic {
+            r#"c8y/s/ds"# => Ok(C8yTopic::SmartRestRequest),
+            r#"c8y/s/us"# => Ok(C8yTopic::SmartRestResponse),
+            err => Err(TopicError::UnknownTopic {
                 topic: err.to_string(),
             }),
         }
     }
 }
-
-impl TryFrom<&str> for IncomingTopic {
-    type Error = MapperTopicError;
+impl TryFrom<&str> for C8yTopic {
+    type Error = TopicError;
 
     fn try_from(value: &str) -> Result<Self, Self::Error> {
         Self::try_from(value.to_string())
     }
 }
 
-impl TryFrom<Topic> for IncomingTopic {
-    type Error = MapperTopicError;
+impl TryFrom<Topic> for C8yTopic {
+    type Error = TopicError;
 
     fn try_from(value: Topic) -> Result<Self, Self::Error> {
         value.name.try_into()
     }
 }
 
 #[derive(Debug, Clone, PartialEq)]
-pub(crate) enum OutgoingTopic {
-    SoftwareListRequest,
-    SoftwareUpdateRequest,
-    SmartRestResponse,
-    RestartRequest,
+pub enum MapperSubscribeTopic {
+    SmartRestRequest,
+    ResponseTopic(ResponseTopic),
 }
 
-impl OutgoingTopic {
-    pub(crate) fn as_str(&self) -> &'static str {
-        match self {
-            Self::SoftwareListRequest => r#"tedge/commands/req/software/list"#,
-            Self::SoftwareUpdateRequest => r#"tedge/commands/req/software/update"#,
-            Self::SmartRestResponse => r#"c8y/s/us"#,
-            Self::RestartRequest => r#"tedge/commands/req/control/restart"#,
+impl TryFrom<String> for MapperSubscribeTopic {
+    type Error = TopicError;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        match value.as_str() {
+            r#"c8y/s/ds"# => Ok(MapperSubscribeTopic::SmartRestRequest),
+            r#"tedge/commands/res/software/list"# => Ok(MapperSubscribeTopic::ResponseTopic(
```

Comment:
It's s a good point to have a enum that encapsulates all the input topic of the mapper
However, the current implementation of the `try_from` method defeats the main point of this PR
that is to encapsulate the agent's topic names inside the agent interface crate.

The `MapperSubscribeTopic ::try_from` implementation must use the `agent_interface::ResponseTopic::try_from` method as well as the SmartRestRequest::try_from` method.